libkbfs/block_types.go

```diff
@@ -116,12 +134,20 @@ func (db *DirBlock) Set(other Block, codec kbfscodec.Codec) {
 	if err != nil {
 		panic("Unable to DirBlock.Set")
 	}
-	*db = *dbCopy
+	db.cacheMtx.Lock()
```

Comment:
This is only meant to protect `cachedEncodedSize`, right?  Can you rearrange this block so the lock only protects that one copy?